lib/resque/pool/killer.rb

```diff
@@ -0,0 +1,41 @@
+require 'open3'
+
+module Resque
+  class Pool
+    class Killer
+      include Logging
+
+      GRACEFUL_SHUTDOWN_SIGNAL=:INT
+
+      def self.run
+        new.run
+      end
+
+      def run
+        my_pid = Process.pid
+        pool_pids = all_resque_pool_processes
+        pids_to_kill = pool_pids.reject{|pid| pid == my_pid}
+        pids_to_kill.each do |pid|
+          log "Pool (#{my_pid}) in kill-others mode: killing pool with pid (#{pid})"
+          Process.kill(GRACEFUL_SHUTDOWN_SIGNAL, pid)
+        end
+      end
+
+
+      def all_resque_pool_processes
+        out, err, status = Open3.capture3("ps ax")
```

Comment:
This ought to always work just fine on Linux/Mac OS X/BSD...  but my paranoia prefers to use `ps -e -o pid= -o command=` (POSIX flags, restricted output columns).  Your current `parse_pids_from_output` should work just fine on either, I think.